### ISSUE TITLE
Self reporting gym competitions with social focus

### ISSUE THREAD

**Issue by scd** [theCrag/website#1485]
Tasks to do before closing:
- [ ] ascents logbook export.
- [ ] add PermitNode to Competition Table
- [ ] add url stub label to Competition Table
- [ ] add Title to Leaderboard table
- [ ] tournament configuration page plus define rigerous status fields.
- [ ] create a separate issue for external site badging

We are a long way off being able to provide software for a proper competition, but we are pretty 

close to being able to do something for a long period (eg over a month) social gym competition.
- [x] filters for leaderboard
- [x] filters should match logbook so that a list of the ascents can be retrieved
- [ ] excel export of the competition logbook
- [x] an ascent score should be flexible, ie based on a formula (eg extra for onsight, less for a repeat)
- [x] Points based on a fixed number divided by the number of ascents - requires updating existing ascents
- [x] Awards for total distance climbed
- [x] One day a week
- [x] Categories
- [ ] Method for configuring a tournament
- [ ] Badging it so that the leaderboard can be accessed with a single click from anywhere the badge is hosted.

Our current leaderboard can work against an total ascent score based on certain filter criteria. The ascent score is recorded against an ascent based on a formula stored on the route. The only valid formula we have now is a constant number, but this could easily be enhanced.

I think there are three areas of work to complete before we can deliver this:
- [x] define and implement a configurable formula for scoring
- [x] define leaderboard filters and which match a logbook
- [x] some sort of tournament configuration. This could be a piece of JSON in a url parameter.

More advanced stuff - public vs private, tournament registration, ascent verification. given the tournaments are more social focus these additional features are not required initially
